hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -198,8 +200,9 @@ public FsVolumeImpl getVolume(final ExtendedBlock b) {
   @Override // FsDatasetSpi
   public Block getStoredBlock(String bpid, long blkid)
       throws IOException {
-    try (AutoCloseableLock lock = lockManager.readLock(LockLevel.BLOCK_POOl,
-        bpid)) {
+    try (AutoCloseableLock lock = lockManager.readLock(LockLevel.DIR,
+        bpid, getReplicaInfo(bpid, blkid).getStorageUuid(),
+        datasetSubLockStrategy.blockIdToSubLock(blkid))) {
```

Comment:
Hi! sir, This query op should be sufficient to obtain a 'POOL' lock, why change it to a 'DIR' lock?

Reply:
@huangzhaobo99 Sir, thanks for your noticing.  The DIR lock is used here to maintain the same mutual exclusion semantics as before.  We should protect the block operations rightly.

Reply:
@hfutatzhanghb Thanks, Got it